lib/tasks/user.rake

```diff
@@ -2,17 +2,26 @@ namespace :usasearch do
   namespace :user do
     desc 'Set site-less users to not_approved'
     task update_approval_status: :environment do
-      User.approved_affiliate.select { |user| user.affiliates.empty? }.each do |user|
-        user.set_approval_status_to_not_approved
-        user.save!
-        Emailer.user_approval_removed(user).deliver_now
-        note = <<~NOTE.squish
-          User #{user.id}, #{user.email}, is no longer associated with any sites,
-          so their approval status has been set to "not_approved".
-        NOTE
+      set_user_not_approved(User.approved_affiliate,
+                            'is no longer associated with any sites',
+                            true)
+    end
+
+    desc 'Set accounts that are not active for more than 90 days to not_approved'
+    task update_inactive_approval_status: :environment do
+      set_user_not_approved(User.not_active, 'has been inactive for 90 days')
+    end
+  end
 
-        Rails.logger.info(note)
-      end
+  def set_user_not_approved(users, message, email = false)
+    users.each do |user|
+      user.set_approval_status_to_not_approved
+      user.save
+      Emailer.user_approval_removed(user).deliver_now if email
+      note = <<~NOTE.squish
+        User #{user.id},  #{user.email}, #{message}, so their approval status has been set to "not_approved".
+      NOTE
+      Rails.logger.info(note)
     end
```

Comment:
This will be refactored again when we have inactive user email setup. That is next sprint. For now it will log and not email yet.